job-service-db/src/main/resources/procedures/reportProgress.sql

```diff
@@ -17,9 +17,51 @@
 /*
  *  Name: report_progress
  *
- *  REMOVED
+ *  Description:
+ *  Reports a task progress to the database
  */
 DROP FUNCTION IF EXISTS report_progress(
     in_task_id VARCHAR(58),
     in_status job_status
 );
+CREATE OR REPLACE PROCEDURE report_progress(
+    in_partition_id VARCHAR(40),
+    in_job_id VARCHAR(58),
+    in_task_id VARCHAR(58),
+    in_task_completion DOUBLE PRECISION
+)
+    LANGUAGE plpgsql
+AS $$
+DECLARE
+    in_status job_status;
+BEGIN
+
+    -- Raise exception if the job identifier has not been specified
+    IF in_job_id IS NULL OR in_job_id = '' THEN
+        RAISE EXCEPTION 'Job identifier has not been specified' USING ERRCODE = '02000'; -- sqlstate no data
+    END IF;
+
+    -- Exit the procedure if task progress is 0
+    IF in_task_completion = 0
+        THEN RETURN;
+    END IF;
+
+    -- Take out an exclusive update lock on the job row
+    PERFORM NULL
+    FROM job j
+    WHERE j.partition_id = in_partition_id
+      AND j.job_id = in_job_id
+        FOR UPDATE;
+
+    IF in_task_completion = 100
+    THEN
+        in_status = 'Completed';
```

Comment:
I don't want this to be able to be used as an alternate to `reportComplete`.  If the caller provides 100% then round down to 99.9%.  If they provide <0% or >100% then raise an exception.

Reply:
Done, but won't we have an issue if we never get that missing 0.01 % ?
Shouldn't we check and re-route from reportJobTasksProgress(final TrackingReportTask trackingReportTask) ?
I added a simple log.warn for now if showing 100 completion but status not "Complete"
{code}
if(trackingReport.estimatedPercentageCompleted==100 && trackingReportStatus!= TrackingReportStatus.Complete){
                    LOG.warn("The task estimatedPercentageCompleted but the status isn't Complete");
                }
{code}